From 7be23b6faa0e50c6bb1168d6676a5e2a4e6edec5 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-7be23b6faa0-20221115T093130Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 15 Nov 2022 10:18:57 +0100
Subject: [PATCH] bisect; remove unused "git-bisect.sh" and ".gitignore" entry
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since fc304fb52f9 (Merge branch 'dd/git-bisect-builtin' into next,
2022-11-14) we've used builtin/bisect.c instead of git-bisect.sh to
implement the "bisect" command. Let's remove the unused leftover
script, and the ".gitignore" entry for the "git-bisect--helper", which
also hasn't been built since fc304fb52f9.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 .gitignore    |  1 -
 git-bisect.sh | 63 ---------------------------------------------------
 2 files changed, 64 deletions(-)
 delete mode 100755 git-bisect.sh

diff --git a/.gitignore b/.gitignore
index cb0231fb401..fe234cfa19a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -20,7 +20,6 @@
 /git-archimport
 /git-archive
 /git-bisect
-/git-bisect--helper
 /git-blame
 /git-branch
 /git-bugreport
diff --git a/git-bisect.sh b/git-bisect.sh
deleted file mode 100755
index f95b8103a9e..00000000000
--- a/git-bisect.sh
+++ /dev/null
@@ -1,63 +0,0 @@
-#!/bin/sh
-
-USAGE='[help|start|bad|good|new|old|terms|skip|next|reset|visualize|view|replay|log|run]'
-LONG_USAGE='git bisect help
-	print this long help message.
-git bisect start [--term-{new,bad}=<term> --term-{old,good}=<term>]
-		 [--no-checkout] [--first-parent] [<bad> [<good>...]] [--] [<pathspec>...]
-	reset bisect state and start bisection.
-git bisect (bad|new) [<rev>]
-	mark <rev> a known-bad revision/
-		a revision after change in a given property.
-git bisect (good|old) [<rev>...]
-	mark <rev>... known-good revisions/
-		revisions before change in a given property.
-git bisect terms [--term-good | --term-bad]
-	show the terms used for old and new commits (default: bad, good)
-git bisect skip [(<rev>|<range>)...]
-	mark <rev>... untestable revisions.
-git bisect next
-	find next bisection to test and check it out.
-git bisect reset [<commit>]
-	finish bisection search and go back to commit.
-git bisect (visualize|view)
-	show bisect status in gitk.
-git bisect replay <logfile>
-	replay bisection log.
-git bisect log
-	show bisect log.
-git bisect run <cmd>...
-	use <cmd>... to automatically bisect.
-
-Please use "git help bisect" to get the full man page.'
-
-OPTIONS_SPEC=
-. git-sh-setup
-
-TERM_BAD=bad
-TERM_GOOD=good
-
-get_terms () {
-	if test -s "$GIT_DIR/BISECT_TERMS"
-	then
-		{
-		read TERM_BAD
-		read TERM_GOOD
-		} <"$GIT_DIR/BISECT_TERMS"
-	fi
-}
-
-case "$#" in
-0)
-	usage ;;
-*)
-	cmd="$1"
-	get_terms
-	shift
-	case "$cmd" in
-	help)
-		git bisect -h ;;
-	*)
-		git bisect--helper "$cmd" "$@" ;;
-	esac
-esac
-- 
2.38.0.1473.g172bcc0511c

